react/features/settings/components/web/audio/AudioSettingsContent.js

```diff
@@ -228,24 +212,36 @@ class AudioSettingsContent extends Component<Props, State> {
      */
     componentWillUnmount() {
         this._componentWasUnmounted = true;
-        this._disposeTrack(this.state.currentMicData);
+        this._disposeTracks(this.state.audioTracks);
```

Comment:
Looks like we are leaking tracks when a device is lost while this mic selection menu is still open since we are disposing the tracks only when the component is unmounted. Will you able to handle that case as well ?

Reply:
<img width="1535" alt="Screen Shot 2020-06-25 at 12 29 47 PM" src="https://user-images.githubusercontent.com/54324652/85760241-a4f98680-b6df-11ea-8922-fca848c47f7a.png">

This is the warning we see in the console when devices get disconnected/connected while the audio popup menu is up.

Reply:
When the devices change we recreate the tracks all over again (see `componentDidUpdate`). 
But before creating the tracks we always dispose (see `setTracks`), so we shouldn't leak.

That warning informs that there are more than 10 listeners attached (which is de default limit).

Reply:
Oh ok. Thanks for explaining !